### PR TITLE
Tracing performance

### PR DESCRIPTION
**Pull request by christo8989**
## Description

Adds virtualization to the tracing tab.

## Screenshot

No style changes.

image

image

## Checklist

[ X ] Displays correctly with both dark and light mode (see useTheme.ts)  
[ - ] Unit/Integration tests added

I added virtualization. I'm not sure how to unit test that.

### CODE REVIEW

**Comment by warrenday** [comment]
Hi @christo8989 

Thanks for adding this. I'm away from my laptop at the moment but will review after the weekend.

Could you explain the benefit this provides over the current tracing view?

**Comment by christo8989** [comment]
@warrenday 

It fixes the performance issues for large sets of data.

**Comment by christo8989** [comment]
Everything is cleaned and ready to be merged.

File: src/components/TracingVisualization/TracingVisualization.tsx [hunk 1]
```diff
@@ -1,33 +1,51 @@
...
-          duration={a.duration}
+          duration={totalTimeNs}
         />
-      ))}
-    </>
+
+        {virtualItems.map(({ key, index, size, start }) => (
+          <TracingVisualizationRow
+            key={key}
+            type={resolvers[index].parentType}
```
Comment (warrenday) [inline src/components/TracingVisualization/TracingVisualization.tsx:34]: `resolvers[index]` is called many times. Could this be assigned to a variable instead.

File: src/components/TracingVisualization/TracingVisualization.tsx [hunk 2]
```diff
@@ -1,33 +1,51 @@
...
+            name={resolvers[index].path.join(".")}
+            total={totalTimeNs}
+            offset={resolvers[index].startOffset}
+            duration={resolvers[index].duration}
+            style={{
+              position: 'absolute',
+              top: 0,
+              left: 0,
+              height: `${size}px`,
```
Comment (warrenday) [inline src/components/TracingVisualization/TracingVisualization.tsx:43]: only height and transform are dynamic here. Could we instead just pass these as props and render the style internally.

Also could we switch `position: 'absolute', top: 0, left: 0` to tailwind classnames to remain consistent with the rest of the app?

**Review by warrenday — APPROVED** [review]
Great work! Thank you

**Pull request merged**